Cesium3DTilesSelection/src/Tile.cpp

```diff
@@ -320,10 +322,9 @@ void Tile::loadContent() {
 
   Tileset& tileset = *this->getTileset();
 
-  std::optional<Future<std::shared_ptr<IAssetRequest>>> maybeRequestFuture =
-      tileset.requestTileContent(*this);
-
-  if (!maybeRequestFuture) {
+  // TODO: rethink some of this logic, in implicit tiling, a tile may be
+  // available but have no content. It may still have children.
+  if (!maybeContentRequest) {
```

Comment:
More generally, I don't quite understand why `loadContent` needed to be changed to receive the request as a parameter, rather than starting the request itself when it deems it to be necessary. Aside from the case above (which is a bug), it looks to be totally equivalent.

Reply:
I wasn't sure if this was the best change, but I did it to set up a way to _not_ request content for implicit tiles that we know not to have content. To decide that, we need the `ImplicitTraversalInfo` which I was considering putting in the `LoadRecord`, so it would be available in `processLoadQueue`. Since it didn't make much sense to further pass the `ImplicitTraversalInfo` into `Tile::loadContent`, I figured it would be better to create a content request in `processLoadQueue` (potentially conditioned on implicit tiling content availability). Actually implementing that seems like a huge pain though, since each callsite of `addTileToLoadQueue` potentially needs the `ImplicitTraversalInfo`. 

The biggest blocker in deciding to go through fully with that change was confusion about the function of `loadContent`. Like you said in the other comment below, `loadContent` does two different things depending on whether or not there is a content request to make. Splitting those into separate functions like you suggest below will help us be more clear on what we're actually doing when there is no content to request. Maybe at that point, the naming of the Tile load states (ContentLoading / ContentLoaded) need to be revisited.